polaris.shopify.com/src/components/ComponentExamples/ComponentExamples.tsx

```diff
@@ -81,61 +85,172 @@ const ComponentExamples = ({examples}: Props) => {
+      },
+      undefined,
+      {shallow: true},
+    );
+  };
+
+  // manage keyboard focus for tabs and tabpanels
+  const examplesLength = examples.length - 1;
+
+  const setSelectedToPreviousTab = () => {
+    if (exampleIndex === 0) {
+      setSelectedTab(examplesLength);
+    } else {
+      const index = exampleIndex - 1;
+      setSelectedTab(index);
+    }
+  };
+
+  const setSelectedToNextTab = () => {
+    if (exampleIndex === examplesLength) {
+      setSelectedTab(0);
+    } else {
+      const index = exampleIndex + 1;
+      setSelectedTab(index);
+    }
+  };
+
+  const setSelectedTab = (index: number) => {
+    exampleIndex = index;
+    updateSelectedExample(index);
+  };
+
+  const onKeyDownHandler = (event: React.KeyboardEvent<HTMLElement>) => {
+    switch (event.key) {
+      case 'ArrowLeft':
+        setSelectedToPreviousTab();
+        break;
+
+      case 'ArrowRight':
+        setSelectedToNextTab();
+        break;
+
+      case 'Home':
+        setSelectedTab(0);
+        break;
+
+      case 'End':
+        setSelectedTab(examplesLength);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const buttonRefs = useRef<(HTMLButtonElement | null)[]>([]);
+
   useEffect(() => {
-    setSelectedIndex(0);
+    buttonRefs.current = buttonRefs.current.slice(0, examples.length);
   }, [examples]);
 
-  return (
-    <Tab.Group
-      defaultIndex={0}
-      selectedIndex={selectedIndex}
-      onChange={setSelectedIndex}
-    >
-      <Tab.List>
-        <div className={styles.ExamplesList} id="examples">
-          {examples.map((example) => {
-            return (
-              <Tab key={example.fileName}>
-                <span>{example.title}</span>
-              </Tab>
-            );
-          })}
-        </div>
-      </Tab.List>
-
-      <Tab.Panels>
-        {examples.map(({fileName, description, code}) => {
-          const exampleUrl = `/examples/${fileName.replace('.tsx', '')}`;
-
-          return (
-            <Tab.Panel key={fileName}>
-              {description ? <Markdown {...description} /> : null}
-              <div className={styles.ExampleFrame}>
-                <iframe
-                  src={exampleUrl}
-                  height={iframeHeight}
-                  onLoad={handleExampleLoad}
-                  id={exampleIframeId}
-                />
-                <div className={className(styles.Buttons, 'light-mode')}>
-                  <CodesandboxButton
-                    className={styles.CodesandboxButton}
-                    code={code}
-                  />
-                </div>
-              </div>
-
-              <Code
-                code={[
-                  {title: 'React', code: code.trim()},
-                  {title: 'HTML', code: htmlCode},
-                ]}
+  useEffect(() => {
+    const currentButtonRef = buttonRefs.current[exampleIndex];
+    if (currentButtonRef) {
+      currentButtonRef.focus();
```

Comment:
without this we lose focus on "page navigation" so we're setting the focus on the selected tab